### PR TITLE
Updated user guides on the option system

### PR DESCRIPTION
**Pull request by jlstevens**
I've finally finished updating the options user guide and splitting out a separate user guide on magics. I think this cuts out a lot of material that is now irrelevant and I hope this is now a clearer story with fewer concepts that people need to worry about.

@jbednar @philippjfr Please review! I'm sure you'll have fixes and suggestions. Once that is done, I'll do a final pass to update to the new name (once we have settled on it).

#### TODO

- [ ] Rename the user guide. Are we happy with 'Applying Customizations' or do we have a better suggestion?

### CODE REVIEW

**Comment by philippjfr** [comment]
I'm generally happy with the content but one thing I was missing is some discussion of inline vs. clone and any issues/benefits associated with that distinctions

**Comment by jlstevens** [comment]
> I'm generally very happy with the content but one thing I was missing is some discussion of inline vs. clone and any issues/benefits associated with that distinctions

There is a little bit of discussion in the section describing `.options`. I suppose I could add a bit more to talk about the cost/benefits of cloning or not cloning...

**Comment by jlstevens** [comment]
@philippjfr @jbednar Added a few sections and applied the rename. I am currently done with this PR so it is ready to review/merge.

File: examples/user_guide/03-Applying_Customization.ipynb [hunk 1]
```diff
@@ -0,0 +1,796 @@
...
+    "\n",
+    "This section of the user guide has described how you can discover and set customization options in HoloViews. Using `hv.help` and the option builders, you should be able to find the options available for any given object you want to display.\n",
+    "\n",
+    "What *hasn't* been explored are some of the facilities HoloViews offers to map the dimensions of your data to style options. This is explored in the [Style Mapping](04-Style_Mapping.ipynb) where you will learn of the `dim` object as well as about the `Cycle` and `Palette` objects.\n",
+    "\n",
+    "The next user guide in this series is [Dimensioned Containers.ipynb](04-Dimensioned_Containers.ipynb) which introduces another core HoloViews concept, namely container objects that can hold elements. Alternatively, if you wish to continue learning about how you can map the data contained in your elements to the screen, you can skip that user guide for now and jump straight to the [Style Mapping](04-Style_Mapping.ipynb) user guide."
+   ]
+  }
+ ],
+ "metadata": {
```
Comment (philippjfr) [inline examples/user_guide/03-Applying_Customization.ipynb]: Metadata not stripped.

**Comment by philippjfr** [comment]
Going to merge, can make fixes in another PR. For now I'll work on updating the numbering and links.

**Pull request merged**